### PR TITLE
feat!: enable dRICH PID in production

### PR DESCRIPTION
**Pull request by c-dilks**
### Briefly, what does this PR introduce?
Include dRICH output collections in default `eicrecon`.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
no

### Does this PR change default behavior?
yes, new output collections

### CODE REVIEW

**Comment by c-dilks** [comment]
First enabling all collections, as a test. We will likely propose including only a subset of them, to avoid storing duplicate information.

**Comment by wdconinc** [comment]
@chchatte92 @c-dilks This would enable the other dRICH collections by default as well. Rebased. I think this will be useful to merge too. Thoughts?

File: src/services/io/podio/JEventProcessorPODIO.cc [hunk 1]
```diff
@@ -51,6 +51,16 @@ JEventProcessorPODIO::JEventProcessorPODIO() {
...
 
+            // DRICH
+            "DRICHRawHits",
+            "DRICHRawHitsAssociations",
+            "DRICHAerogelTracks",
+            "DRICHGasTracks",
+            "DRICHMergedTracks",
+            "DRICHAerogelIrtCherenkovParticleID",
+            "DRICHGasIrtCherenkovParticleID",
+            // "DRICHMergedIrtCherenkovParticleID",
```
Comment (c-dilks) [inline src/services/io/podio/JEventProcessorPODIO.cc:62]: Let's drop the redundant (merged) ones.
```suggestion
            "DRICHAerogelIrtCherenkovParticleID",
            "DRICHGasIrtCherenkovParticleID",
```

File: src/services/io/podio/JEventProcessorPODIO.cc [hunk 2]
```diff
@@ -75,6 +83,7 @@ JEventProcessorPODIO::JEventProcessorPODIO() {
             "ReconstructedChargedParticleAssociations",
             "ReconstructedSeededChargedParticles",
             "ReconstructedSeededChargedParticleAssociations",
+            // "ReconstructedChargedParticleIDs",
```
Comment (c-dilks) [inline src/services/io/podio/JEventProcessorPODIO.cc:86]: Do we need to include these? The one-to-many relation from `ReconstructedChargedParticles` points to them already, but I can't remember if this collection needs to be explicitly included in the output.
```suggestion
            "ReconstructedChargedParticleIDs",
```

**Comment by c-dilks** [comment]
> This would enable the other dRICH collections by default as well.

This will be very useful for high-stats studies from the September production. At minimum we need hits and track projections.

**Comment by wdconinc** [comment]
Cool, can someone turn this from draft into full PR? I can approve and we can be done with this quickly.

**Review by wdconinc — DISMISSED** [review]

**Review by wdconinc — APPROVED** [review]

**Pull request merged**